### PR TITLE
Block positions range handling minor code cleanup

### PR DESCRIPTION
**Pull request by findepi**

### CODE REVIEW

File: presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java [hunk 1]
```diff
@@ -61,6 +59,13 @@ static void checkValidRegion(int positionCount, int positionOffset, int length)
         }
     }
 
+    static void checkValidPosition(int positionCount, int position)
```
Comment (sopel39) [inline presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java]: I think it's more natural to pass `position` first then `positionCount` as if asking: if `position` is correct in array with `positionCount` elements?

File: presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java [hunk 2]
```diff
@@ -34,11 +34,11 @@ private BlockUtil()
     {
     }
 
-    static void checkValidPositionsArray(int[] positions, int offset, int length)
+    static void checkValidArrayOffsets(int[] array, int offset, int length)
```
Comment (sopel39) [inline presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java]: `checkValidArrayOffsets` is not very meaningful too. Maybe `checkValidArrayRange` or `checkValidSubarray`?

Comment (findepi) [reply presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java]: `checkValidArrayRange` sounds good

File: presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java (hunk 1 above)
Comment (findepi) [reply presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java]: I agree, but i wanted to be consistent with `BlockUtil#checkValidRegion`. Do you think we can change order arguments in that method too?

File: presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java [hunk 3]
```diff
@@ -34,11 +34,11 @@ private BlockUtil()
     {
     }
 
-    static void checkValidPositionsArray(int[] positions, int offset, int length)
+    static void checkValidArrayRange(int[] array, int offset, int length)
```
Comment (nezihyigitbasi) [inline presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java]: Why not just `checkArrayRange`? A check already implies that we are checking that the array range is valid.

Maybe we should rename similarly to Guava, they have a similar method named `checkPositionIndexes`.

Comment (findepi) [reply presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java]: Good point! I think in our everything-is-a-position world, `checkArrayRange` provides less ambiguity, I'll go with that.

**Comment by findepi** [comment]
> The commit message Remove BlockUtil.checkValidPositions(int[] positions, .., positionCount) is too long, you can remove the parameters I guess.

I now it's not the shortest, but otherwise `checkValidPositions` it's looks almost the same as `checkValidPosition` (a sibling commit mentions that).

**Comment by findepi** [comment]
AC

File: presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java (hunk 1 above)
Comment (sopel39) [reply presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java]: Not that important

File: presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java [hunk 4]
```diff
@@ -266,7 +266,7 @@ public BlockEncoding getEncoding()
     @Override
     public Block copyPositions(int[] positions, int offset, int length)
     {
-        checkValidPositions(positions, offset, length, positionCount);
+        checkValidPositionsArray(positions, offset, length);
```
Comment (sopel39) [inline presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java]: missing `checkValidPosition(positionCount, position);` below?

File: presto-spi/src/main/java/com/facebook/presto/spi/block/VariableWidthBlock.java [hunk 5]
```diff
@@ -132,7 +132,7 @@ public void retainedBytesForEachPart(BiConsumer<Object, Long> consumer)
     @Override
     public Block copyPositions(int[] positions, int offset, int length)
     {
-        checkValidPositions(positions, offset, length, positionCount);
+        checkValidPositionsArray(positions, offset, length);
```
Comment (sopel39) [inline presto-spi/src/main/java/com/facebook/presto/spi/block/VariableWidthBlock.java]: missing `checkValidPosition(positionCount, position);` below?

File: presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java (hunk 4 above)
Comment (findepi) [reply presto-spi/src/main/java/com/facebook/presto/spi/block/DictionaryBlock.java]: `getId(position)` below does that

File: presto-spi/src/main/java/com/facebook/presto/spi/block/VariableWidthBlock.java (hunk 5 above)
Comment (findepi) [reply presto-spi/src/main/java/com/facebook/presto/spi/block/VariableWidthBlock.java]: same, every position is validated

File: presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java (hunk 1 above)
Comment (findepi) [reply presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java]: @sopel39 i updated the signature

Comment (sopel39) [reply presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java]: Still LGTM :)

**Pull request merged**